sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -198,6 +195,30 @@ case class InsertIntoHiveTable(
       }
     }
 
+    table.catalogTable.bucketSpec match {
+      case Some(bucketSpec) =>
+        // We can not populate bucketing information for Hive tables as Spark SQL has a different
+        // implementation of hash function from Hive.
+        // Hive native hashing will be supported after SPARK-17495. Until then, writes to bucketed
+        // tables are allowed only if user does not care about maintaining table's bucketing
+        // ie. both "hive.enforce.bucketing" and "hive.enforce.sorting" are set to false
+
+        val enforceBucketingConfig = "hive.enforce.bucketing"
+        val enforceSortingConfig = "hive.enforce.sorting"
+
+        val message = s"Output Hive table ${table.catalogTable.identifier} is bucketed but Spark" +
+          "currently does NOT populate bucketed output which is compatible with Hive."
+
+        if (hadoopConf.get(enforceBucketingConfig, "false").toBoolean ||
+          hadoopConf.get(enforceSortingConfig, "false").toBoolean) {
```

Comment:
Are the default values (`false`) for these two configs safe? If user doesn't aware of it, it could insert non compatible data into bucketed Hive table.